src/manifest.c

```diff
@@ -1146,21 +1059,49 @@ struct list *consolidate_files(struct list *files)
 		file1 = list->data;
 		file2 = next->data;
 
+		/* If the filenames are different, nothing to consolidate, just move on. */
 		if (strcmp(file1->filename, file2->filename)) {
 			list = next;
 			continue;
-		} /* from here on, file1 and file2 have a filename match */
+		}
+
+		/* From here on, file1 and file2 have a filename match. */
 
-		/* (case 1) C and C'               : choose file1
-		 * this is the most common case, so test it first */
-		if (!file1->is_deleted && !file2->is_deleted && hash_equal(file1->hash, file2->hash)) {
-			/* always drop the untracked file if there is a tracked file */
+		/* If both files are present, this is the most common case. */
+		if (!file1->is_deleted && !file2->is_deleted) {
+
+			/* If the hashes don't match, we have an inconsistency in the
+			 * manifest. Exclude both from the list. */
+			if (!hash_equal(file1->hash, file2->hash)) {
+				tmp = next->next;
+				list_free_item(list, NULL);
+				list_free_item(next, NULL);
+				list = tmp;
+				telemetry(TELEMETRY_INFO,
```

Comment:
I think a `TELEMETRY_WARN` or `TELEMETRY_CRIT` severity would be more appropriate, since this condition would reflect a serious bug in manifest creation.

Reply:
Changed to CRIT.